### ISSUE TITLE
UnderKeyboardLayoutConstraint memory leak

### ISSUE THREAD

**Issue by danionescu** [exchangegroup/under-keyboard-ios#10]
Hello,

There is a memory leak in UnderKeyboardLayoutConstraint.swift, due to a circular reference to self, found here: https://github.com/evgenyneu/UnderKeyboard/blob/master/UnderKeyboard/UnderKeyboardLayoutConstraint.swift#L22-L23 . This leak also causes the `viewToAnimate` to leak.

It can easily be fixed by replacing the above two lines with
```swift
    keyboardObserver.willAnimateKeyboard = { [weak self] height in
        self?.keyboardWillAnimate(height)
    }
    keyboardObserver.animateKeyboard = { [weak self] height in
        self?.animateKeyboard(height)
    }
```

I am happy to submit a pull request, if needed.

Thank you,
Dan

**Comment by evgenyneu** [comment]
Thanks for finding this, you are right. A pull request is welcome. 👍

**Comment by danionescu** [comment]
Done, thank you!

**Comment by evgenyneu** [comment]
Thank you, I've released your changes in version 10.0.0.
